src/services/helpers.ts

```diff
@@ -9,14 +9,47 @@ export class ValidationError extends Error {
   }
 }
 
+export class TimeoutError extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = "TimeoutError";
+  }
+}
+
+async function processTimeout<T>(
+  ms: number,
+  promise: Promise<T>,
+  errorMessage?: string
+): Promise<T> {
+  return new Promise<T>(async (resolve, reject) => {
+    setTimeout(() => {
+      reject(new TimeoutError(errorMessage || "Request timed out"));
+    }, ms);
+    resolve(await promise);
+  });
+}
+
 export async function fetchWithValidator<T, O, I>(
   validator: Type<T, O, I>,
   requestInfo: RequestInfo,
-  init?: RequestInit
+  init?: RequestInit,
+  timeout: boolean | number = false
 ): Promise<T> {
-  const response = await fetch(requestInfo, init);
+  let response;
+  if (timeout) {
+    const timeoutVal =
+      typeof timeout === "number" ? timeout : process.env.API_CALL_TIMEOUT;
```

Comment:
Make sure to default `process.env.API_CALL_TIMEOUT` too, since it may not be defined

Reply:
i think we don't have to worry cos timeout is default false without the "?" 
<img width="358" alt="Screenshot 2020-07-16 at 2 09 37 PM" src="https://user-images.githubusercontent.com/38589870/87633491-102dfa00-c76e-11ea-8ee2-7d709bf4748f.png">

Reply:
Oh right, there's no path that will get a possibly undefined process env